src/tools/compiletest/src/runtest.rs

```diff
@@ -3027,11 +3027,17 @@ impl<'test> TestCx<'test> {
         const PREFIX: &str = "MONO_ITEM ";
         const CGU_MARKER: &str = "@@";
 
+        // Some MonoItems can contain {closure@/path/to/checkout/tests/codgen-units/test.rs}
+        // To prevent the current dir from leaking, we just replace the entire path to the test
+        // file with TEST_PATH.
         let actual: Vec<MonoItem> = proc_res
             .stdout
             .lines()
             .filter(|line| line.starts_with(PREFIX))
-            .map(|line| str_to_mono_item(line, true))
+            .map(|line| {
+                line.replace(&self.testpaths.file.display().to_string(), "TEST_PATH").to_string()
+            })
```

Comment:
Discussion (not blocking): I think this is fine for `codegen-units` tests, but ui tests normalize paths in their output with placeholders like `$SRC_DIR`, `$DIR`, `$COMPILER_DIR`. Should we follow that for consistency?

https://github.com/rust-lang/rust/blob/730d5d4095a264ef5f7c0a0781eea68c15431d45/src/tools/compiletest/src/runtest.rs#L4322-L4328

Reply:
(unrelated remark I need to breakup / tidy up `runtest.rs` because the 4000 lines of run test logic is a pain every time I try to look up something)

Reply:
Yeah. We should probably apply all the path normalizations to all tests to be honest. It can be hard to remember how little code sharing there is between test suites.

Reply:
#128952